TPP.Persistence.MongoDB/Repos/BadgeRepo.cs

```diff
@@ -102,6 +103,7 @@ public async Task<List<Badge>> FindAllByCustom(string? userId = null, PkmnSpecie
                 filter &= Builders<Badge>.Filter.Eq(b => b.Form, form);
             if (source != null)
                 filter &= Builders<Badge>.Filter.Eq(b => b.Source, source);
+            filter &= Builders<Badge>.Filter.Eq(b => b.Shiny, shiny); // always assigned, defaults to false
```

Comment:
Hmm, maybe queries do become awkward at places without a migration after all. Does this work properly for `false`? Because it might not, since null != false :/

Reply:
assuming this works the way I think it does

cm.MapProperty(b => b.Shiny).SetElementName("shiny")
                    .SetDefaultValue(false)
                    .SetIgnoreIfDefault(true);

when it loads things from the database it should interpret missing fields and null values as false.

Reply:
yeah but this is a filter for a query and it might get serialized to mongodb syntax as-is. Can you write a test to verify this works for nulls? You can look at e.g. `BankTest#can_handle_null_transaction_type` to see how to insert raw bson data in a test.
If you don't have the time or don't wanna do it for any reason let me know, then I can test it.

Reply:
I should be able to get to it soon™